File: lib/models/Complaints.js
import mongoose from "mongoose";

const complaintSchema = new mongoose.Schema({
    user :{type: mongoose.Schema.Types.ObjectId, ref: "User"},
    product: { type: mongoose.Schema.Types.ObjectId, ref: "Product" },
    message: String,
    status: { type: String, enum: ["open", "in-progress", "resolved"], default: "open" },
    createdAt: { type: Date, default: Date.now }
});

const Complaint = mongoose.models.Complaint || mongoose.model("Complaint", complaintSchema);
export default Complaint;

File: backend/models/Complaints.py
from mongoengine import Document, StringField, ReferenceField, DateTimeField
from datetime import datetime

class Complaint(Document):
    user = ReferenceField("users", required=True)       
    product = ReferenceField("products", required=True) 
    message = StringField()
    status = StringField(
        choices=["open", "in-progress", "resolved"],
        default="open"
    )
    createdAt = DateTimeField(default=datetime.utcnow)

    meta = {"collection": "complaints"}  
